routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
so, the context gets passed into run, and then we create a goprocess. We set that goprocess to be closed when the context gets cancelled, and then we derive a context from that process to pass to the queryFuncs. it seems a little redundant.